numpy/random/generator.pyx

```diff
@@ -3141,6 +3141,11 @@ cdef class Generator:
         replacement (or the sample space is infinite). As the sample space
         becomes large, this distribution approaches the binomial.
 
+        The arguments `ngood` and `nbad` each must be less than `10**9`. For
```

Comment:
Is there a reparameterization that can be suggested for larger values?  I woud presum that as long ngood/ntotal and nbad/ntotal are both moderaely far from 0, when ntotal is large than a N(nsample*p, nsample*p*(1-p))*(ntotal-nsample)/(ntotal-1)) is a good approx.  

Not sure this is worth it.

Reply:
I don't think it is worthwhile to do this now.   Let's wait for someone with a legitimate need for such an enhancement before spending time implementing it.

Reply:
Yes. I wasn't actually suggesting code to do this -- rather telling users to use a normal approximation if they really want to sample in this region. 

This is a very minor point, and probably not worth making. 

A related pioint is that the strict check against HYPERGEOM_MAX should be written into the doc string for ngood and nbad as requirements, .e.g, ngood must be less than 1e9.

Reply:
The requirement that the arguments be less then `10**9` is well-hidden at the end of the Notes section. 😄   I'll add that requirement to the description of the parameters in the Parameters section.

Reply:
> ... telling users to use a normal approximation if they really want to sample in this region.

The binomial distribution is probably a simpler approximation in this case.  I'll add a note suggesting `binomial(n=nsample, p=ngood/(ngood + nbad))` as an alternative.